// üß™ Enunciado ‚Äì Exerc√≠cio 4: Simular erro de rede com uma flag
// Use seu useAxios, mas:

// Adicione uma flag booleana chamada forcarErro (por padr√£o false)

// No buscarDados, se forcarErro for true, dispare um erro manual:

// if (forcarErro) throw new Error('Erro simulado!')
// No componente App, adicione um bot√£o chamado ‚ÄúSimular Erro‚Äù

// Ao clicar nesse bot√£o, altere a flag forcarErro pra true e recarregue os dados

// Mostre corretamente a mensagem de erro e o bot√£o de ‚ÄúTentar novamente‚Äù


import Button from "./Button"
import useAxios from "./useAxios"

const App = () => {
  const { error, dados, loading, recarregar, forcarErro, simule } = useAxios()

  if (loading) return <p>Carregando...</p>

  if (error) return (
    <div>
      <p style={{ color: 'red' }}>{error}</p>
      <Button onClick={recarregar}>Tentar novamente</Button>
    </div>
  )

  if (!dados) return <p>Nenhum dado carregado.</p>


  return (
    <div>
      {/* Leitura dev: Se os dados existem (foram carregados) e o carregamento terminou, ent√£o mostre a lista. */}
      {dados && !loading && (
        <ul>
          {dados && dados.map((nome) => (
            <li key={nome.id}>{nome.username}</li>
          ))}
        </ul>
      )}
      {!loading && (
        <div style={{ display: 'flex', gap: '.5rem' }}>
          <Button onClick={recarregar}>Recarregar</Button>
          <Button onClick={simule}>For√ßar erro</Button>
        </div>
      )}

    </div>

  )
}

export default App
